### ISSUE TITLE
`searchsortedfirst` slower for ranges than for vectors

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#44296]
I'm surprised to see that `searchsorted` is faster when looking for a random value in a 10-element `Vector{Float64}` than when looking for it in a range. `LinRange` is noticeably faster than `StepRangeLen`, but still slower than `Vector{Float64}`. Is that expected?
```julia
julia> x = rand(10_000);

julia> @btime searchsortedfirst.(Ref($(collect(0:0.1:1))), x);
  196.085 μs (5 allocations: 78.25 KiB)

julia> @btime searchsortedfirst.(Ref(0:0.1:1), x);
  334.781 μs (5 allocations: 78.30 KiB)

julia> @btime searchsortedfirst.(Ref(LinRange(0, 1, 11)), x);
  247.149 μs (5 allocations: 78.28 KiB)
```

This affects the performance of histograms in StatsBase.

**Comment by N5N3** [comment]
For such small size, it should be normal that a cached `Array` is faster than float `range`? (especally for `StepRangeLen` with `TwicePrecision`)

**Comment by vtjnash** [comment]
Yes, different accuracies are expected to cost differently, particularly when your microbenchmark can fit in L1 cache.
